include/boost/wave/util/cpp_macromap.hpp

```diff
@@ -1046,12 +1108,19 @@ bool adjacent_stringize = false;
             // simply copy the original argument (adjacent '##' or '#')
 #if BOOST_WAVE_SUPPORT_VARIADICS_PLACEMARKERS != 0
                 if (is_ellipsis) {
-                position_type const &pos = (*cit).get_position();
-
-                    impl::trim_sequence_left(arguments[i]);
-                    impl::trim_sequence_right(arguments.back());
-                    BOOST_ASSERT(boost::wave::need_variadics(ctx.get_language()));
-                    impl::replace_ellipsis(arguments, i, expanded, pos);
+                    position_type const &pos = (*cit).get_position();
+                    if (i < arguments.size()) {
+
+                        impl::trim_sequence_left(arguments[i]);
+                        impl::trim_sequence_right(arguments.back());
+                        BOOST_ASSERT(boost::wave::need_variadics(ctx.get_language()));
+                        impl::replace_ellipsis(arguments, i, expanded, pos);
+                    } else {
+                        BOOST_ASSERT(i == arguments.size());
+                        // no argument supplied; insert placemarker
+                        *std::back_inserter(expanded) =
+                            typename ContainerT::value_type(T_PLACEMARKER, "\xA7", pos);
+                    }
```

Comment:
Should this `else` be C++20 only? Or does it fix a bug?

Reply:
Sounds reasonable. I will make the change.